x-pack/plugins/security_solution/public/management/components/endpoint_responder/lib/console_commands_definition.ts

```diff
@@ -421,5 +442,68 @@ export const getEndpointConsoleCommands = ({
     });
   }
 
+  // `execute` is currently behind feature flag
+  // planned for 8.8
+  if (isExecuteEnabled) {
+    consoleCommands.push({
+      name: 'execute',
+      about: getCommandAboutInfo({
+        aboutInfo: i18n.translate('xpack.securitySolution.endpointConsoleCommands.execute.about', {
+          defaultMessage: 'Execute a command on the host',
+        }),
+        isSupported: doesEndpointSupportCommand('execute'),
+      }),
+      RenderComponent: ExecuteActionResult,
+      meta: {
+        endpointId: endpointAgentId,
+        capabilities: endpointCapabilities,
+        privileges: endpointPrivileges,
+      },
+      exampleUsage: 'execute --command "ls -al" --timeout 2s --comment "Get list of all files"',
+      exampleInstruction: ENTER_OR_ADD_COMMENT_ARG_INSTRUCTION,
+      validate: capabilitiesAndPrivilegesValidator,
+      mustHaveArgs: true,
+      args: {
+        command: {
+          required: true,
+          allowMultiples: false,
+          about: i18n.translate(
+            'xpack.securitySolution.endpointConsoleCommands.execute.args.command.about',
+            {
+              defaultMessage: 'The command to execute',
+            }
+          ),
+          validate: (argData) => {
+            return emptyArgumentValidator(argData);
+          },
+        },
+        timeout: {
+          required: false,
+          allowMultiples: false,
+          about: i18n.translate(
+            'xpack.securitySolution.endpointConsoleCommands.execute.args.timeout.about',
+            {
+              defaultMessage:
+                'The timeout in units of time (h for hours, m for minutes, s for seconds) for the endpoint to wait for the script to complete. Example: 37m. If not given, it defaults to 4 hours.',
+            }
+          ),
+          validate: executeTimeoutValidator,
```

Comment:
You still need this validation for parsing out the timeout value, but you should also define the `mustHaveValue` property here adn set it to `non-empty-string`

Reply:
997bde19c8da4e7798793f8fcd7041902412a6f1